### PR TITLE
[fix](memleak) avoid memleak due to race condition

### PR DESCRIPTION
**Pull request by dataroaring**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h [hunk 1]
```diff
@@ -51,6 +51,9 @@ namespace doris::vectorized {
 struct AggregateFunctionHLLData {
     HyperLogLog dst_hll {};
 
+    AggregateFunctionHLLData() = default;
+    ~AggregateFunctionHLLData() = default;
```
Comment (zhangstar333) [inline be/src/vec/aggregate_functions/aggregate_function_hll_union_agg.h:55]: In my opinion，this is seems generated by the compiler automatically？
Where does this memory leak come from

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.11 seconds
 stream load tsv:          423 seconds loaded 74807831229 Bytes, about 168 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230427033630_clickbench_pr_136142.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**